homeassistant/components/ollama_conversation/config_flow.py

```diff
@@ -0,0 +1,169 @@
+"""Config flow for Ollama conversation integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import sys
+from types import MappingProxyType
+from typing import Any
+
+import httpx
+import ollama
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    OptionsFlow,
+)
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.selector import (
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    ObjectSelector,
+    TemplateSelector,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import (
+    CONF_MAX_HISTORY,
+    CONF_MODEL,
+    CONF_MODEL_OPTIONS,
+    CONF_PROMPT,
+    DEFAULT_PROMPT,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    MAX_HISTORY_NO_LIMIT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.URL)
+        ),
+        vol.Required(CONF_MODEL): cv.string,
+        vol.Required(
+            CONF_PROMPT, description={"suggested_value": DEFAULT_PROMPT}
+        ): TemplateSelector(),
+        vol.Optional(CONF_MODEL_OPTIONS): ObjectSelector(),
```

Comment:
Ieuw

Reply:
We should not ask this on initial creation of the config entry.

Reply:
Removed model options for now